src/components/scenes/SendScene2.tsx

```diff
@@ -839,6 +860,23 @@ const SendComponent = (props: Props) => {
     if (onBack != null) onBack()
   })
 
+  React.useEffect(() => {
+    if (timeoutHandler.current != null) {
+      clearTimeout(timeoutHandler.current)
+    }
+    if (expireDate != null) {
+      const nowSeconds = new Date().getTime() / 1000
+      const timeoutSeconds = expireDate.getTime() / 1000
+      const expireS = Math.max(Math.round(timeoutSeconds - nowSeconds), 0)
+      setExpireSeconds(expireS)
+      if (expireS <= 0) {
+        setError(new Error(s.strings.send_address_expired_error_message))
+      }
+      timeoutHandler.current = setTimeout(() => setTick(tick + 1), 500)
+      return () => clearTimeout(timeoutHandler.current)
+    }
+  }, [expireDate, tick])
+
```

Comment:
I wonder if we can factor out a `CountdownTile` component, that handles the timer internally and presents an `onDone` handler for when we reach the end. That way, we don't re-render the huge SendScene2 component 2x per second. React will only need to diff the tile itself.

The SendScene2 would continue to provide the `onDone` logic, of course, but that only fires once after the 10 minutes are up.

Reply:
good idea. done!